tensilelite/Tensile/ClientWriter.py

```diff
@@ -213,31 +214,29 @@ def getBuildClientLibraryScript(buildPath, libraryLogicPath, cxxCompiler):
     callCreateLibraryCmd += " --short-file-names"
 
   if globalParameters.get("AsmDebug", False):
-    callCreateLibraryCmd += " --asm-debug"
+    callCreateLibraryCmd += [" --asm-debug",]
 
   if globalParameters["KeepBuildTmp"]:
-    callCreateLibraryCmd += " --keep-build-tmp"
+    callCreateLibraryCmd += [" --keep-build-tmp"]
 
   callCreateLibraryCmd += " --architecture=" + globalParameters["Architecture"]
   callCreateLibraryCmd += " --code-object-version=" + globalParameters["CodeObjectVersion"]
   callCreateLibraryCmd += " --cxx-compiler=" + cxxCompiler
   callCreateLibraryCmd += " --library-format=" + globalParameters["LibraryFormat"]
 
-  callCreateLibraryCmd += " %s" % libraryLogicPath
-  callCreateLibraryCmd += " %s" % buildPath #" ../source"
-  callCreateLibraryCmd += " %s\n" % globalParameters["RuntimeLanguage"]
+  callCreateLibraryCmd += [" %s" % libraryLogicPath]
+  callCreateLibraryCmd += [" %s" % buildPath] #" ../source"
+  callCreateLibraryCmd += [" %s\n" % globalParameters["RuntimeLanguage"]]
 
-  runScriptFile.write(callCreateLibraryCmd)
+  return callCreateLibraryCmd
```

Comment:
There's some issues here and causes tox tests failure on gfx942:
 - the return type was changed to `list`, but the caller treats it as string(refer to L101).
 - mixing concatenating `list` and `str`, which may generate unexpected result if you want to simply join them.

You can run tox test locally to reproduce the error.